packages/platform-core/src/config/config_controller.ts

```diff
@@ -0,0 +1,130 @@
+import os from 'os';
+import path from 'path';
+import fs from 'fs';
+
+/**
+ * Used to interact with config file based on OS.
+ */
+export class ConfigController {
+  dirPath: string;
+  configFilePath: string;
+  _store: Record<string, unknown>;
+
+  /**
+   * Initializes common paths to telemetry configs.
+   */
+  constructor(
+    private readonly projectName = 'amplify',
+    private readonly configFileName = 'config.json'
+  ) {
```

Comment:
Is this thread safe? e.g. if there are two instances who are doing
```ts
const inst1 = new ConfigController();
const inst2 = new ConfigController();
inst1.set('a.b', 1);
inst2.set('c.d', 1);
inst1.set('a.b', 2);
```
I'm guessing that this will wipe out `c.d` key and this is for a single threaded execution and no race condition. It would be much better if we have smaller/specific config files and a factory that vends out configController instances based on the fileName

Reply:
Another option here is to make ConfigController a singleton (should probably do that regardless)

Reply:
It has to be a static instance nevertheless that factory would generate, but we should still create an instance specific for a specific file instead of a general purpose config file

```ts
Factory {
  getInstance(config) {
    if(instanceExists(config)) {
      return existingInstance
    }
    create newInstance(config);
  }
}
```